NBitcoin/Transaction.cs

```diff
@@ -180,12 +180,15 @@ void SetNull()
 		{
 			return !(a == b);
 		}
+
 		public override bool Equals(object? obj)
 		{
-			var item = obj as OutPoint;
-			if (object.ReferenceEquals(null, item))
-				return false;
-			return item == this;
+			return Equals(obj as OutPoint);

```

Comment:
```suggestion
			return obj is OutPoint op && Equals(op);
```

Reply:
It is unnecessary because in case it `obj` is not an `OutPoint` instance, after casting it is `null` and will be evaluated as false by the `==` operator.

Reply:
Indeed. I didn't know that, I thought it throws.

Reply:
Btw this could be a good opportunity for @yahiheb to improve our IEquitable implementations based on this information.